terraform/modules/elastic_user/main.tf

```diff
@@ -0,0 +1,37 @@
+variable "service" {
+  type = string
+}
+
+variable "roles" {
+  type = list(string)
+}
+
+resource "aws_secretsmanager_secret" "username" {
+  name        = "elasticsearch/catalogue_api/${var.service}/username"
+  description = "Config secret populated by Terraform"
+}
+
+resource "aws_secretsmanager_secret" "password" {
+  name        = "elasticsearch/catalogue_api/${var.service}/password"
+  description = "Config secret populated by Terraform"
+}
+
+resource "aws_secretsmanager_secret_version" "username" {
+  secret_id     = aws_secretsmanager_secret.username.id
+  secret_string = var.service
+}
+
+resource "aws_secretsmanager_secret_version" "password" {
+  secret_id     = aws_secretsmanager_secret.username.id
+  secret_string = random_password.password.result
+
+  provisioner "local-exec" {
+    command = "python3 ../modules/elastic_user/create_user.py ${var.service} ${random_password.password.result} ${join(",", var.roles)}"
```

Comment:
Will the roles get updated when the var changes? I'm not sure that a change in `var.roles` would retrigger this provisioner.

Reply:
Nope, that's a piece I am trying to work out. In the description I am thinking we could have an `update_user` script, but would prefer to have it happen automatically.

Maybe a `null_resource` or something with that as a dep that runs an update_user_roles script... I'll have a play.